chain/client/src/chunk_validation.rs

```diff
@@ -40,6 +42,9 @@ pub struct ChunkValidator {
     epoch_manager: Arc<dyn EpochManagerAdapter>,
     network_sender: Sender<PeerManagerMessageRequest>,
     runtime_adapter: Arc<dyn RuntimeAdapter>,
+    /// Maps hash of chunk currently being validated into marker of validation
+    /// completeness.
+    chunks_in_validation: Arc<RwLock<HashMap<ChunkHash, Arc<OnceCell<()>>>>>,
```

Comment:
Why not have a boolean instead of `Arc<OnceCell<()>>`? Or better, just a HashSet?

Reply:
Because `Arc<OnceCell<()>>` abstracts away the waiting until the value has been set. Infinite loop (spinlock) also works but I'm less comfortable with it, it looks expensive.